addons/account/models/account_reconcile_model.py

```diff
@@ -238,79 +247,62 @@ def _get_write_off_move_lines_dict(self, st_line, move_lines=None):
+                continue
+
+            if line.amount_type == 'percentage':
+                line_balance = residual_balance * (line.amount / 100.0)
+            elif line.amount_type == "regex":
+                match = re.search(line.amount_from_label_regex, st_line.name)
                 if match:
-                    line_balance = copysign(float(re.sub(r'\D' + self.decimal_separator, '', match.group(1)).replace(self.decimal_separator, '.')), remaining_balance)
+                    line_balance = copysign(float(re.sub(r'\D' + line.decimal_separator, '', match.group(1)).replace(line.decimal_separator, '.')) * (1 if balance > 0.0 else -1), balance)
                 else:
                     line_balance = 0
             else:
-                line_balance = self.second_amount * (1 if remaining_balance > 0.0 else -1)
-
-            second_writeoff_line = {
-                'name': self.second_label or st_line.name,
-                'account_id': self.second_account_id.id,
-                'analytic_account_id': self.second_analytic_account_id.id,
-                'analytic_tag_ids': [(6, 0, self.second_analytic_tag_ids.ids)],
+                line_balance = line.amount * (1 if residual_balance > 0.0 else -1)
+            writeoff_line = {
+                'name': line.label or st_line.name,
+                'account_id': line.account_id.id,
+                'analytic_account_id': line.analytic_account_id.id,
+                'analytic_tag_ids': [(6, 0, line.analytic_tag_ids.ids)],
                 'debit': line_balance > 0 and line_balance or 0,
                 'credit': line_balance < 0 and -line_balance or 0,
                 'reconcile_model_id': self.id,
             }
-            new_aml_dicts.append(second_writeoff_line)
-
-            if self.second_tax_ids:
-                second_writeoff_line['tax_ids'] = [(6, None, self.second_tax_ids.ids)]
-                tax = self.second_tax_ids
+            new_aml_dicts.append(writeoff_line)
+            if for_widget:
+                self._make_line_for_widget(writeoff_line)
+                writeoff_line['to_check'] = line.model_id.to_check
+                writeoff_line['__focus'] = True
+
+            residual_balance -= line_balance
+            if line.tax_ids:
+                writeoff_line['tax_ids'] = [(6, None, line.tax_ids.ids)]
+                tax = line.tax_ids
                 # Multiple taxes with force_tax_included results in wrong computation, so we
                 # only allow to set the force_tax_included field if we have one tax selected
-                if self.force_second_tax_included:
+                if line.force_tax_included:
                     tax = tax[0].with_context(force_price_include=True)
-                new_aml_dicts += self._get_taxes_move_lines_dict(tax, second_writeoff_line)
+                new_aml_dicts += self._get_taxes_move_lines_dict(tax, writeoff_line)
+
+        if for_widget:
```

Comment:
mysterious code. Dubious. Me no like :confused:

Reply:
this last piece of code could be placed in `get_reconciliation_dict_for_widget`, if my previous comment is right, which would allow to remove the `for_widget` parameter (did I already mention I do not like it?)

Reply:
I am also putting `get_reconciliation_dict_for_widget` in enterprise as the widget has moved. No need to see that in community...

Reply:
good idea